### PR TITLE
[Core] Adding doxygen to IO and ModelPartIO

### PR DESCRIPTION
**Pull request by loumalouomega**
Also correcting some wrongly assigned virtual/override functions

### CODE REVIEW

File: kratos/includes/io.h [hunk 1]
```diff
@@ -117,139 +117,306 @@ class KRATOS_API(KRATOS_CORE) IO
...
     ///@{
 
+    /**
+     * @brief This method reads one node
+     * @param rThisNode The node to be read
+     */
     virtual bool ReadNode(NodeType& rThisNode)
     {
-        KRATOS_ERROR <<  "Calling base class member. Please check the definition of derived class." << std::endl;
+        KRATOS_ERROR << "Calling base class method (ReadNode). Please check the definition of derived class." << std::endl;
```
Comment (philbucher) [inline kratos/includes/io.h:126]: is this necessary? Afaik KRATOS_ERROR tells you already in which method it threw doesn't it?

**Review by philbucher — COMMENTED** [review]
one question otherwise I will approve

File: kratos/includes/io.h (hunk 1 above)
Comment (loumalouomega) [reply kratos/includes/io.h:126]: It is not harmful

**Review by philbucher — APPROVED** [review]

**Comment by loumalouomega** [comment]
Thanks

**Pull request merged**